packages/spacecat-shared-gpt-client/src/clients/firefall-client.js

```diff
@@ -107,15 +107,22 @@ export default class FirefallClient {
     this.log.debug(`${message}: took ${duration}ms`);
   }
 
-  async #submitPrompt(body, path) {
+  /**
+   * Submit a prompt to the Firefall API.
+   * @param body The body of the request.
+   * @param path The Firefall API path.
+   * @param imsOrgId An optional IMS Org ID, to override the config default.
+   * @returns {Promise<unknown>}
+   */
+  async #submitPrompt(body, path, imsOrgId = undefined) {
```

Comment:
Instead of of an extra parameter maybe the third parameter should be an object of overrides?

Reply:
Where does `this.config.imsOrg` gets its IMS Org value from?

Reply:
Could this logic be moved to `FirefallClient.createFrom`?

```
    const {
      FIREFALL_API_ENDPOINT: apiEndpoint,
      IMS_CLIENT_ID: imsClientId,
      IMS_ORG: imsOrg,
      FIREFALL_API_KEY: apiKey,
      FIREFALL_API_POLL_INTERVAL: pollInterval = 2000,
      FIREFALL_API_CAPABILITY_NAME: capabilityName = 'gpt4_32k_completions_capability',
    } = context.env;

   const firefallImsOrg = imsOrg || imsClientId;

    return new FirefallClient({
      apiEndpoint,
      apiKey,
      capabilityName,
      imsClient,
      imsOrg: firefallImsOrg,
      pollInterval,
    }, log);
```

Reply:
> Where does this.config.imsOrg gets its IMS Org value from?
From `IMS_CLIENT_ID: imsOrg,` line 53.

Reply:
> Could this logic be moved to FirefallClient.createFrom?
`createFrom` and `context` seems to dealing with ENV variables (from AWS secrets) and Spacecat injected values.
Since our IMS_ORG used in the FF call isn't part of that, it doesn't seem to fit in `createFrom()`.

If a GPTClient called `fetchChatCompletion()` twice, it might want to use 2 different IMS ORG ids, but has to use the same context.

Reply:
Are you picturing this:
```
async #submitPrompt(body, path, overrides = {}) {
 ...
  const headers = {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${apiAuth}`,
      'x-api-key': this.config.apiKey,
      'x-gw-ims-org-id': imsOrgId ?? this.config.imsOrg,
      ...overrides,
    }
```
I am thinking the use here of "_this.config.imsOrg (i.e. IMS_CLIENT_ID)_" as an IMS ORG ID is actually incorrect (or at least misleading) and it would be the only exception ever made.  I could make it overrides if you want.

Reply:
> > Could this logic be moved to FirefallClient.createFrom?
> > `createFrom` and `context` seems to dealing with ENV variables (from AWS secrets) and Spacecat injected values.
> > Since our IMS_ORG used in the FF call isn't part of that, it doesn't seem to fit in `createFrom()`.
> 
> If a GPTClient called `fetchChatCompletion()` twice, it might want to use 2 different IMS ORG ids, but has to use the same context.

Is this an actual issue or a hypothetical issue? If the ims org is coming from the api key which is part of the context why would you want to change the ims org during the same request?

Reply:
> Are you picturing this:
> 
> ```
> async #submitPrompt(body, path, overrides = {}) {
>  ...
>   const headers = {
>       'Content-Type': 'application/json',
>       Authorization: `Bearer ${apiAuth}`,
>       'x-api-key': this.config.apiKey,
>       'x-gw-ims-org-id': imsOrgId ?? this.config.imsOrg,
>       ...overrides,
>     }
> ```
> 
> I am thinking the use here of "_this.config.imsOrg (i.e. IMS_CLIENT_ID)_" as an IMS ORG ID is actually incorrect (or at least misleading) and it would be the only exception ever made. I could make it overrides if you want.

I am now preferring the possibility of needing any overrides at this level. I think it just makes the API more confusing having different places where header values can be set.

Reply:
Hypothetical I suppose.
Since this is shared in all of spacecat, I'd prefer not to assume how some other service would use it.  I guess they could create a new GPTClient if need be.
Why do you think it should be embedded in the client, instead of just during the call where it is used?  The latter feels more flexible but if you are sure, I can change it.

Reply:
We should get some more historical context form @solaris007. At first glance it seems confusing to allow only img org to be set two different ways. When creating a `FirefallClient` I would expect all the `config` values it needs would come from the context. It would be nice if all the `FirefallClient` methods worked the same way. `fetchChatCompletion` shouldn't need to set it img org differently than `fetchCapabilityExecution` does. So the question that needs to be anwered is why does `fetchCapabilityExecution` use the ims client id as its ims org? Is this because there was no ims org to use? Is this a bug?